core/src/main/scala/ai/lum/odinson/state/State.scala

```diff
@@ -53,11 +55,25 @@ trait State {
 
 object State {
 
-  def apply(config: Config, indexSearcher: OdinsonIndexSearcher, indexDir: Directory): State = {
+  def apply(config: Config, index: OdinsonIndex, indexDir: Directory): State = {
+    val provider = config.apply[String]("odinson.state.provider")
+    val state = provider match {
+      // The SQL state needs an IndexSearcher to get the docIds from the
+      case "sql"    => SqlState(config, index, Some(index.directory))
+      case "file"   => FileState(config)
+      case "memory" => MemoryState(config)
+      case "mock"   => MockState
+      case _        => throw new Exception(s"Unknown state provider: $provider")
+    }
+
+    state
+  }
+
+  def apply(config: Config, index: OdinsonIndex): State = {
     val provider = config.apply[String]("odinson.state.provider")
```

Comment:
It seems like these two could be combined.

Reply:
@kwalcock this may have been a miss on my part, i might have kept the old method there while i implemented the new one, let me take a look.